histogrammatching.py

```diff
@@ -0,0 +1,132 @@
+__doc__ = """\
+HistogramMatching 
+================
+**HistogramMatching** it manipulates the pixel intesity values an input image and matches
+them to the histogram of a reference image. It can be used as a way to normalize intensities 
+across different images or, or different frames of the same image. It allows you to choose 
+which frame to use as the reference. 
+
+
+============ ============ ===============
+Supports 2D? Supports 3D? Respects masks?
+============ ============ ===============
+YES          YES          NO
+============ ============ ===============
+
+References
+^^^^^^^^^^
+(`link <http://paulbourke.net/miscellaneous/equalisation/>`__)
```

Comment:
What do you think about adding the skimage tutorial here too? https://scikit-image.org/docs/stable/auto_examples/color_exposure/plot_histogram_matching.html

Reply:
Yes good idea, I will add that